### PR TITLE
Refactor docs for DI and Application sections

### PR DESCRIPTION
**Pull request by wsargent**
This is a big documentation PR, but I think it's worth it.

The "Dependency Injection" sections in Working with Java / Working with Scala are under the advanced section.  At the same time, we're deprecating GlobalSettings.

This PR creates an "Application" section in the docs.  It contains the ErrorHandling, HttpFilters, and so on, that is `play.api.Application` or `play.Application` based, so there are some pages moved out from "http".  I also added a general Application page and slapped deprecation notices for `GlobalSettings` on everything.

I have also moved the dependencyinjection section from the "advanced" area to the main area.   What's left under "advanced" is "embedding", "extending", and "routing".  I think those are fine where they are. 

Things not changed:

There are lots of Java sections that still have reference to a Global class, notably JavaLogging.md.  I have left that as is.

This also shows that Java does not have an HttpRequestHandler section that reflects the Scala one.  I have moved JavaHttpRequestHandler.md to JavaActionCreator.md in anticipation of https://groups.google.com/d/topic/play-framework-dev/lcYyNwejLSo/discussion because it does not handle routeRequest, but I'm going to leave a Java HttpRequestHandler page to another PR.

### CODE REVIEW

File: documentation/manual/working/javaGuide/main/application/JavaInterceptors.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <!--- Copyright (C) 2009-2016 Typesafe Inc. <http://www.typesafe.com> -->
 # Intercepting requests
 
+> **NOTE**: The `GlobalSettings` class is deprecated in 2.5.x.  Please see the [[Removing `GlobalSettings`|GlobalSettings]] page for how to migrate away from GlobalSettings.
```
Comment (gmethvin) [inline documentation/manual/working/javaGuide/main/application/JavaInterceptors.md]: We should note here what the correct way of intercepting requests is.

File: documentation/manual/working/javaGuide/main/dependencyinjection/JavaDependencyInjection.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!--- Copyright (C) 2009-2016 Typesafe Inc. <http://www.typesafe.com> -->
-# Dependency Injection
+# Dependency Injection with Guice
```
Comment (gmethvin) [inline documentation/manual/working/javaGuide/main/dependencyinjection/JavaDependencyInjection.md]: :+1: The docs are assuming Guice so we might as well be explicit about it up front.

**Comment by wsargent** [comment]
Woo!

**Comment by gmethvin** [comment]
Also some additional doc changes in #5500

**Pull request merged**